src/Fixture/Factory/ProductBundleExampleFactory.php

```diff
@@ -0,0 +1,111 @@
+{
+    /**
+     * @var ProductBundleCreatorInterface
+     */
+    private $productBundleCreator;
+
+    /**
+     * @var ProductRepositoryInterface
+     */
+    private $productRepository;
+
+    /**
+     * @var OptionsResolver
+     */
+    private $optionsResolver;
+
+    public function __construct(
+        ProductBundleCreatorInterface $productBundleCreator,
+        ProductRepositoryInterface $productRepository
+    ) {
+        $this->productBundleCreator = $productBundleCreator;
+        $this->productRepository = $productRepository;
+        $this->optionsResolver = new OptionsResolver();
+        $this->configureOptions($this->optionsResolver);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setDefined('productCode')
+            ->setAllowedTypes('productCode', 'string')
+            ->setDefined('slots')
+            ->setAllowedTypes('slots', 'array');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $options = []): ProductBundleInterface
+    {
+        $options = $this->optionsResolver->resolve($options);
+
+        /** @var ProductInterface $product */
+        $product = $this->productRepository->findOneByCode($options['productCode']);
+
+        Assert::notNull($product, sprintf('Bundle product %s could not be found', $options['productCode']));
+        $productBundleCreator = $this->productBundleCreator->createProductBundle($product->getName(), $product);
+
+        foreach ($options['slots'] as $slot) {
+            $slotProducts = [];
+            foreach ($slot['productCodes'] as $productCode) {
+                $slotProduct = $this->productRepository->findOneByCode($productCode);
+                Assert::notNull($slotProduct, sprintf('Slot product %s could not be found', $productCode));
+                $slotProducts[] = $slotProduct;
+            }
+            $slotOptions = $this->createSlotOptions($slot['options']);
+            $productBundleCreator->addSlot($slot['name'], $slotOptions, $slotProducts);
+        }
+
+        /** @var ProductBundleInterface $productBundle */
+        $productBundle = $productBundleCreator->getProductBundle();
+        $productBundle->setCode($product->getCode());
+
+        return $productBundle;
+    }
+
+    /**
+     * @param string[] $rawSlotOptions
+     */
+    private function createSlotOptions(array $rawSlotOptions = []): ProductBundleSlotOptionsInterface
+    {
+        $slotOptions = new ProductBundleSlotOptions();
+        foreach ($rawSlotOptions as $optionName => $optionValue) {
+            $setter = 'set' . ucfirst($optionName);
+
+            Assert::methodExists($slotOptions, $setter, sprintf('Setter %s for ProductBundleSlotOptions is not defined', $setter));
+
+            if (method_exists($slotOptions, $setter = 'set' . ucfirst($optionName))) {
+                $slotOptions->$setter($optionValue);
```

Comment:
Just as a note, there is also the possibility to add additional options with the 
[ProductBundleSlotOptionsInterface::addOptionMethod](https://github.com/solutionDrive/SyliusProductBundlesPlugin/blob/84d57030be3646700c18a98d4c3d1df413625c48/src/Service/Options/OptionsInterface.php#L14) in an else-branch if that is a thing that is needed.